.github/workflows/build.yml

```diff
@@ -0,0 +1,83 @@
+---
+name: "node-opendds CI"
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+      - MigrateToGithubActions
+  schedule:
+    - cron: '10 0 * * 0'
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        node_version:
+          - 10
+        opendds_branch:
+          # - master
+          # - latest-release
+          - branch-DDS-3.14
+        m:
+          - {os: ubuntu-latest, dds_security: 1}
+          - {os: macos-latest, dds_security: 0}
+    runs-on: ${{ matrix.m.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: 'Set environment variables'
+        run: |-
+          echo "ACE_ROOT=$GITHUB_WORKSPACE/ACE_TAO/ACE" >> $GITHUB_ENV
+          echo "TAO_ROOT=$GITHUB_WORKSPACE/ACE_TAO/TAO" >> $GITHUB_ENV
+          echo "DDS_ROOT=$GITHUB_WORKSPACE/OpenDDS" >> $GITHUB_ENV
+          echo "MPC_ROOT=$GITHUB_WORKSPACE/MPC" >> $GITHUB_ENV
+          echo "NVM_DIR=$GITHUB_WORKSPACE/nvm" >> $GITHUB_ENV
+          echo "NAN_ROOT=$GITHUB_WORKSPACE/node_modules/nan" >> $GITHUB_ENV
+          echo "npm_config_devdir=$GITHUB_WORKSPACE/opendds-node-gyp-devdir" >> $GITHUB_ENV
+          echo "LD_LIBRARY_PATH=$GITHUB_WORKSPACE/ACE_TAO/ACE/lib:$GITHUB_WORKSPACE/OpenDDS/lib:$GITHUB_WORKSPACE/tools/v8stubs:$LD_LIBRARY_PATH" >> $GITHUB_ENV
+      - name: 'Clone OpenDDS'
+        run: |-
+          git clone --depth 1 git://github.com/DOCGroup/MPC.git
+          git clone --depth 1 -b Latest_Micro git://github.com/DOCGroup/ACE_TAO.git
+          git clone --depth 1 -b ${{ matrix.opendds_branch }} git://github.com/objectcomputing/OpenDDS.git
+          if [ ${{ matrix.m.dds_security }} == 1 ]; then
+            echo 'Install ssl and xerces'
+            sudo apt-get update
```

Comment:
move these installs to a separate step... you can make the step conditional (similar to this if statement).